tls/s2n_alerts.c

```diff
@@ -245,64 +228,16 @@ int s2n_process_alert_fragment(struct s2n_connection *conn)
     return 0;
 }
 
-int s2n_queue_writer_close_alert_warning(struct s2n_connection *conn)
-{
-    POSIX_ENSURE_REF(conn);
-
-    uint8_t alert[2];
-    alert[0] = S2N_TLS_ALERT_LEVEL_WARNING;
-    alert[1] = S2N_TLS_ALERT_CLOSE_NOTIFY;
-
-    struct s2n_blob out = { 0 };
-    POSIX_GUARD(s2n_blob_init(&out, alert, sizeof(alert)));
-
-    /* If there is an alert pending, do nothing */
-    if (s2n_stuffer_data_available(&conn->writer_alert_out)) {
-        return S2N_SUCCESS;
-    }
-
-    if (!s2n_alerts_supported(conn)) {
-        return S2N_SUCCESS;
-    }
-
-    POSIX_GUARD(s2n_stuffer_write(&conn->writer_alert_out, &out));
-
-    return S2N_SUCCESS;
-}
-
-static int s2n_queue_reader_alert(struct s2n_connection *conn, uint8_t level, uint8_t error_code)
-{
-    POSIX_ENSURE_REF(conn);
-
-    uint8_t alert[2];
-    alert[0] = level;
-    alert[1] = error_code;
-
-    struct s2n_blob out = { 0 };
-    POSIX_GUARD(s2n_blob_init(&out, alert, sizeof(alert)));
-
-    /* If there is an alert pending, do nothing */
-    if (s2n_stuffer_data_available(&conn->reader_alert_out)) {
```

Comment:
I'm not sure if this  is important or could ever happen, but one possible difference in the new implementation is that I think it would replace a queued alert with a new alert, rather than keep an alert that's already queued.

Reply:
It can't happen, because all the code that queues alerts is inaccessible after a fatal error closes the connection. But I can add a check for that just in case.